src/slashers/VetoableSlasher.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {IStrategy} from "eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import {SlasherBase} from "./SlasherBase.sol";
+
+contract VetoableSlashing is SlasherBase {
+    struct SlashingRequest {
+        address operator;
+        uint32 operatorSetId;
+        IStrategy[] strategies;
+        uint256 wadToSlash;
+        string description;
+        uint256 requestTimestamp;
+        SlashingStatus status;
+    }
+
+    uint256 public constant VETO_PERIOD = 3 days;
+    address public vetoCommittee;
+    uint256 public nextRequestId;
+    mapping(uint256 => SlashingRequest) public slashingRequests;
+
+    event SlashingRequested(
+        uint256 indexed requestId,
+        address indexed operator,
+        uint32 indexed operatorSetId,
+        uint256 wadToSlash,
+        string description
+    );
+
+    event SlashingRequestCancelled(uint256 indexed requestId);
+
+    modifier onlyVetoCommittee() {
+        require(msg.sender == vetoCommittee, "VetoableSlashing: caller is not the veto committee");
+        _;
+    }
+
+    function initialize(address _serviceManager, address _vetoCommittee) external virtual initializer {
+        __SlasherBase_init(_serviceManager);
+        vetoCommittee = _vetoCommittee;
+    }
+
+    function queueSlashingRequest(
+        address operator,
+        uint32 operatorSetId,
+        IStrategy[] memory strategies,
+        uint256 wadToSlash,
+        string memory description
+    ) external virtual {
+        _queueSlashingRequest(operator, operatorSetId, strategies, wadToSlash, description);
+    }
+
+    function cancelSlashingRequest(uint256 requestId) external virtual onlyVetoCommittee {
```

Comment:
vetoSlashingRequest

Reply:
Request, Slash, and Cancel are more generic. Cancellation is just happening through a veto committee in this case